### PR TITLE
Implement site search

### PR DESCRIPTION
**Pull request by settermjd**
This PR implements site search into the documentation, powered by Lunr. There is an as yet unresolved bug generating the documentation with a Docker container, but the documentation can be generated from the command-line successfully.

### CODE REVIEW

**Comment by settermjd** [comment]
@tboerger, wondering if you could have a look at the Dockerfile to see what I might have missed?

**Comment by settermjd** [comment]
@tboerger, this works without the Docker container. Time-permitting I can implement changes to get the Docker container to build the docs.

File: site.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 site:
   title: ownCloud Documentation
+  url: http://localhost:5000
```
Comment (tboerger) [inline site.yml]: Hardcoded to localhost doesn't look good.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "version": "10.0.10",
+  "description": "The ownCloud core documentation, created with Antora and AsciiDoc",
+  "directories": {
+    "doc": "docs"
+  },
+  "dependencies": {
+    "@antora/cli": "^1.1.1",
+    "@antora/site-generator-default": "^1.1.1",
+    "antora-lunr": "v0.1.0",
+    "lunr": "v2.3.3",
```
Comment (tboerger) [inline package.json]: This is a peer dependency and not a direct dependency, right?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "write-good": "^0.13.1"
```
Comment (tboerger) [inline package.json]: Where is that required?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+# Enable Lunr site search integration and generation of a Lunr search index by default 
+ENV DOCSEARCH_ENABLED true
+ENV DOCSEARCH_ENGINE lunr
+
+WORKDIR /antora
+
+COPY package* /antora/
+
+RUN apk add --no-cache build-base libressl-dev libcurl libgit2-dev python && \
```
Comment (tboerger) [inline Dockerfile]: Where are these packages required?

File: site.yml (hunk 1 above)
Comment (settermjd) [reply site.yml]: True. I got stuck here and need to fix this.

File: package.json (hunk 2 above)
Comment (settermjd) [reply package.json]: I had to do [some reading](https://lexi-lambda.github.io/blog/2016/08/24/understanding-the-npm-dependency-model/) (_always good fun_ 👍) to know exactly what you meant. If I understand you correct, yes, it's a peer dependency. And [antora-lunr's package.json](https://github.com/Mogztter/antora-lunr/blob/master/package.json) shows that it's already a dependency. So it seems that I've added it unnecessarily. _Will fix._

File: package.json (hunk 3 above)
Comment (settermjd) [reply package.json]: That slipped in. Will remove.

File: Dockerfile (hunk 4 above)
Comment (settermjd) [reply Dockerfile]: I installed them as the build failed without them.

File: site.yml (hunk 1 above)
Comment (settermjd) [reply site.yml]: I changed this to something akin to a Docker environment variable.

**Pull request merged**